tools/report_generators/cli.py

```diff
@@ -0,0 +1,291 @@
+        debug=debug,
+    )
+
+    if debug:
+        germany_regions = germany_regions[:10]
+
+    gre.create_html_reports(germany_regions)
+
+    gre.create_markdown_index_page()
+
+
+def generate_reports_usa(*, workers, kernel_name, wwwroot, force, disable_pbar, debug):
+    _ = oscovida.fetch_cases_US()
+    _ = oscovida.fetch_deaths_US()
+
+    #  TODO: The get_x_list methods should be part of Reporter class
+    states = oscovida.get_US_region_list()
+
+    usre = ReportExecutor(
+        Reporter=USAReport,
+        wwwroot=wwwroot,
+        expiry_hours=2,
+        attempts=3,
+        workers=workers,
+        force=force,
+        disable_pbar=disable_pbar,
+        debug=debug,
+    )
+
+    if debug:
+        states = states[:10]
+
+    usre.create_html_reports(states)
+
+    usre.create_markdown_index_page()
+
+
+def generate_markdown_all_regions(
+    *, workers, kernel_name, wwwroot, force, disable_pbar, debug
+):
+    arre = ReportExecutor(Reporter=AllRegions, wwwroot=wwwroot)
+
+    arre.create_markdown_index_page()
+
+
+def generate(*, region, workers, kernel_name, wwwroot, force, disable_pbar, debug):
+    mapping = {
+        "countries": generate_reports_countries,
+        "germany": generate_reports_germany,
+        "usa": generate_reports_usa,
+        "all-regions-md": generate_markdown_all_regions,
+    }
+
+    mapping[region](
+        workers=workers,
+        kernel_name=kernel_name,
+        wwwroot=wwwroot,
+        disable_pbar=disable_pbar,
+        force=force,
+        debug=debug,
+    )
+
+
+@click.command()
+@click.option(
+    "--regions",
+    "-r",
+    type=click.Choice(
+        ["countries", "germany", "usa", "all-regions-md", "all"], case_sensitive=False
+    ),
+    multiple=True,
+    help="Region(s) to generate reports for.",
+)
+@click.option(
+    "--workers",
+    default="auto",
+    help="Number of workers to use, 'auto' uses nproc-2, set to 1 or False to "
+    "use a single process.",
+)
+@click.option("--wwwroot", default="./wwwroot", help="Root directory for www content.")
+@click.option(
+    "--create-wwwroot",
+    default=False,
+    is_flag=True,
+    help="Create wwwroot directory if it does not exist.",
+)
+@click.option(
+    "--kernel-name", default="", help="Create wwwroot directory if it does not exist."
+)
+@click.option(
+    "--disable-pbar",
+    default=False,
+    is_flag=True,
+    help="Disable progress bar, print logging output instead.",
+)
+@click.option(
+    "--log-level",
+    default="WARNING",
+    type=click.Choice(["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"]),
+    help="Log level.",
+)
+@click.option("--log-file", default=None, help="Log file.")
+@click.option(
+    "--force",
+    default=False,
+    is_flag=True,
+    help="Force notebook re-execution even if recently executed.",
+)
+@click.option(
+    "--debug",
+    default=False,
+    is_flag=True,
+    help="Enable debug mode, uses a small subset of regions.",
+)
+def cli(
```

Comment:
more descriptive name may be here?

Reply:
Added in a docstring to the function which describes what it's for, is that okay?

Reply:
thanks. Yes that's ok.